metadata_templates.py

```diff
@@ -88,9 +173,65 @@ def metadata_templates_collection_gather(rule_args, callback, rei):
                 schemas.append(j)
             except Exception as e:
                 callback.writeLine('serverLog', '{}'.format(type(e)))
+        elif thetype == 'irods':
+            # get length of iRODS full logical path
+            ret = callback.msiObjStat(schema, irods_types.RodsObjStat())
+            objstat = ret['arguments'][1]
+            # get schema contents from iRODS full logical path
+            ret = callback.msiDataObjOpen(schema, 0)
+            fd = ret['arguments'][1] # iRODS file descriptor
+            ret = callback.msiDataObjRead(fd, objstat.objSize, irods_types.BytesBuf())
+            buf = ret['arguments'][2] # bytesbuf
+            byteslist = buf.get_bytes() # schema contents as list of bytes as integers
+            callback.msiDataObjClose(fd, 0)
+            # convert to json object
+            j = json.loads(bytes(byteslist).decode("utf-8"))
+            # add to schemas array
+            schemas.append(j)
+        elif thetype == 'local':
+            # get contents
+            with open(schema, 'r') as f:
+                content = f.read()
+                # convert to json object
+                j = json.loads(content)
+                # add to schemas array
+                schemas.append(j)
```

Comment:
This looks identical to the implementation for data objects. Consider making it a function.

Reply:
https://github.com/irods-contrib/irods_working_group_metadata_templates/issues/41